src/utils/array.ts

```diff
@@ -70,12 +70,15 @@ export function resizeArray<T = unknown>(
 		}, []);
 }
 
-export function firstItem<T = unknown>(items: T | T[]): T {
+/**
+ * Return the first non null/undefined value of an array
+ */
+export function firstValueItem<T = unknown>(items: T | T[]): T {
 	if (!Array.isArray(items)) {
 		return items;
 	}
 	for (const item of items) {
-		if (item !== undefined) {
+		if (item !== undefined && item !== null) {
 			return item;
 		}
 	}
```

Comment:
I would prefer this code : 
```ts
/**
 * Return the first non null/undefined value of an array
 */
export function firstValueItem<T = unknown>(items: T | T[]) {
	const getItem = (item: T | undefined): T | undefined => item ?? undefined;

	return Array.isArray(items)
		? items.map(getItem).find((item) => item !== undefined)
		: getItem(items);
}
```

Reply:
I prefer to do an early return, if there is lots of items its quicker to exit sooner and I think the loop code is clearer (but it's your call)

Reply:
Yep your correct i replace map by find